common/k8s/src/middleware/metadata.rs

```diff
@@ -35,79 +35,57 @@ enum Error {
 pub struct K8sMetadata {
     metadata: Mutex<HashMap<(String, String), PodMetadata>>,
     api: Api<Pod>,
-    runtime: Mutex<Option<Runtime>>,
 }
 
 // TODO refactor to use kube-rs Reflector instead of manually managing hashmap
 impl K8sMetadata {
-    pub fn new() -> Result<Self, K8sError> {
-        let runtime = match Builder::new_multi_thread()
```

Comment:
This issue was luckily caught by the k8s integration test (we shouldn't create a new runtime from a runtime context :) )

Reply:
Nice work around, I'd love to refactor the Middlewares so they can be spawned on the main.rs executor rather than running in their own threads, but that's definitely out of scope for this PR